contribs/src/main/java/com/netflix/conductor/contribs/queue/kafka/KafkaObservableQueue.java

```diff
@@ -0,0 +1,230 @@
+ */
+package com.netflix.conductor.contribs.queue.kafka;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.KafkaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.conductor.core.config.Configuration;
+import com.netflix.conductor.core.events.queue.Message;
+import com.netflix.conductor.core.events.queue.ObservableQueue;
+import com.netflix.conductor.core.execution.ApplicationException;
+
+import rx.Observable;
+import rx.Observable.OnSubscribe;
+
+/**
+ * Reads the properties with prefix 'kafka.producer.', 'kafka.consumer.'
+ * and 'kafka.' from the provided configuration. Initializes a producer 
+ * and consumer based on the given value. Queue name is driven from the 
+ * workflow. 
+ * @author preeth
+ *
+ */
+public class KafkaObservableQueue implements ObservableQueue {
+
+  private static final Logger logger = LoggerFactory.getLogger(KafkaObservableQueue.class);
+
+  private static final String QUEUE_TYPE = "kafka";
+
+  private static final String KAFKA_PRODUCER_PREFIX = "kafka.producer.";
+
+  private static final String KAFKA_CONSUMER_PREFIX = "kafka.consumer.";
+
+  private static final String KAFKA_PREFIX = "kafka.";
+
+  private final String queueName;
+
+  private final int pollTimeInMS;
+
+  private final int longPollTimeout;
+
+  private final int pollCount;
+
+  private final KafkaProducer<String, String> producer;
+
+  private final KafkaConsumer<String, String> consumer;
+
+  @Inject
+  public KafkaObservableQueue(String queueName, Configuration config) {
+
+    this.queueName = queueName;
+    this.pollTimeInMS = config.getIntProperty("workflow.dyno.queues.pollingInterval", 100);
+    this.pollCount = config.getIntProperty("workflow.dyno.queues.pollCount", 10);
+    this.longPollTimeout = config.getIntProperty("workflow.dyno.queues.longPollTimeout", 1000);
+    // Init Kafka producer and consumer properties
+    Properties producerProps = new Properties();
+    Properties consumerProps = new Properties();
+    consumerProps.put("group.id", queueName + "_group");
+    String serverId = config.getServerId();
+    consumerProps.put("client.id", queueName + "_consumer_"+serverId);
+    producerProps.put("client.id", queueName + "_producer_"+serverId);
+    Map<String, Object> configMap = config.getAll();
```

Comment:
There are some set of properties that are required to create the consumer, like the key deserializer and the value deserializer, These properties are not defined in the config nor there is a validation made to check the existence of them before creating the consumer

Reply:
It was not defined as the module is configurable. So currently put the mandatory configurations in a commented mode in the config,
Also added validation for the mandatory configuration for producer and consumer.
Please suggest.